sql/expression/logic.go

```diff
@@ -37,11 +37,19 @@ func JoinAnd(exprs ...sql.Expression) sql.Expression {
 	case 0:
 		return nil
 	case 1:
+		if exprs[0] == nil {
+			return nil
+		}
```

Comment:
(minor) seems like these three lines could be omitted and still have the same effect of returning nil.